irods_docker_files/run_tests_in_parallel.py

```diff
@@ -107,21 +107,45 @@ def main():
 
     run_pool = Pool(processes=int(args.test_parallelism))
 
-    containers = [{'test_name': docker_cmd['test_name'], 'proc': run_pool.apply_async(docker_cmds_utilities.run_command_in_container, (docker_cmd['run_cmd'], docker_cmd['exec_cmd'], docker_cmd['stop_cmd'], docker_cmd['container_name'], docker_cmd['alias_name'], docker_cmd['database_container'], docker_cmd['database_type'], docker_cmd['network_name'],),{'test_type': docker_cmd['test_type'],})} for docker_cmd in docker_cmds_list]
-
-    container_error_codes = [{'test_name': c['test_name'], 'error_code': c['proc'].get()} for c in containers]
-
-    print(container_error_codes)
+    containers = [{
+        'test_name': docker_cmd['test_name'],
+        'proc': run_pool.apply_async(
```

Comment:
Anything printed to stdout/err by the function provided to `apply_async` needs to acquire a lock or be buffered and printed later. This is necessary so that asynchronous output does not interleave.

Reply:
After discussion, the solution is to have each container maintain its own file.